pallets/gear/src/lib.rs

```diff
@@ -941,6 +955,52 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Resumes a previously paused program
+        ///
+        /// The origin must be Signed and the sender must have sufficient funds to
+        /// transfer value to the program.
+        ///
+        /// Parameters:
+        /// - `program_id`: id of the program to resume.
+        /// - `memory_pages`: program memory before it was paused.
+        /// - `value`: balance to be transferred to the program once it's been resumed.
+        ///
+        /// - `ProgramResumed(H256)` in the case of success.
+        #[frame_support::transactional]
+        #[pallet::weight(<T as Config>::WeightInfo::resume_program(memory_pages.values().map(|p| p.len() as u32).sum()))]
+        pub fn resume_program(
+            origin: OriginFor<T>,
+            program_id: H256,
+            memory_pages: BTreeMap<u32, Vec<u8>>,
+            value: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let account = ensure_signed(origin)?;
+
+            ensure!(!value.is_zero(), Error::<T>::ResumeProgramNotEnoughValue);
```

Comment:
This is a place I'm not certain with.

I suggest that it is nice to have some constant representing minimal rent for several blocks and check that `value` is greater or equal that constant. However it could be introduced in the corresponding task.